spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/repository/JdbcDataflowJobExecutionDao.java

```diff
@@ -38,17 +44,27 @@
 	private final NamedParameterJdbcTemplate jdbcTemplate;
 	private final String tablePrefix;
 
+	//	/**
+	//	 * SQL statements for removing the Step Execution Context.
+	//	 */
+	//	private static final String  SQL_DELETE_BATCH_STEP_EXECUTION_CONTEXT =
+
+	/**
+	 * SQL statements for retrieving Step Execution Ids.
+	 */
+	private static final String SELECT_STEP_EXECUTION_IDS =
+			"SELECT SEC.STEP_EXECUTION_ID AS STEP_EXECUTION_ID " +
+			"FROM %PREFIX%STEP_EXECUTION_CONTEXT SEC " +
+			"JOIN %PREFIX%STEP_EXECUTION SE ON SE.STEP_EXECUTION_ID = SEC.STEP_EXECUTION_ID " +
+			"WHERE SE.JOB_EXECUTION_ID in (:jobExecutionIds)";
+
 	/**
 	 * SQL statements for removing the Step Execution Context.
 	 */
 	private static final String  SQL_DELETE_BATCH_STEP_EXECUTION_CONTEXT =
 			"DELETE FROM %PREFIX%STEP_EXECUTION_CONTEXT " +
 			"WHERE STEP_EXECUTION_ID " +
-			"IN ( " +
-			"SELECT SEC.STEP_EXECUTION_ID " +
-			"FROM %PREFIX%STEP_EXECUTION_CONTEXT SEC " +
-			"JOIN %PREFIX%STEP_EXECUTION SE ON SE.STEP_EXECUTION_ID = SEC.STEP_EXECUTION_ID " +
-			"WHERE SE.JOB_EXECUTION_ID in (:jobExecutionIds))";
+			"IN (:stepExecutionIds)";
```

Comment:
Will we have issues with step execution size in this in clause.   I've seen jobs with hundreds of steps.

Reply:
Created  https://github.com/spring-cloud/spring-cloud-dataflow/issues/3348